Assets/Scripts/UI/Overlay/OverlayMap.cs

```diff
@@ -0,0 +1,486 @@
+﻿using UnityEngine;
+using System.Collections;
+using System;
+using System.Collections.Generic;
+using MoonSharp.Interpreter;
+
+/// <summary>
+/// This game object manages a mesh+texture+renderer+material that is
+/// used to superimpose a semi-transparent "overlay" to the map.
+/// </summary>
+[RequireComponent(typeof(MeshFilter))]
+[RequireComponent(typeof(MeshRenderer))]
+public class OverlayMap : MonoBehaviour {
+
+    public Dictionary<string, OverlayDescriptor> overlays;
+
+    /// <summary>
+    /// Starting left corner (x,y) and z-coordinate of mesh and (3d left corner)
+    /// </summary>
+    public Vector3 leftBottomCorner = new Vector3(-0.5f, -0.5f, 1f);
+    /// <summary>
+    /// Transparency of overlay
+    /// </summary>
+    [Range(0,1)]
+    public float transparency = 0.4f;
+    /// <summary>
+    /// Update interval (0 for every Update, inf for never)
+    /// </summary>
+    public float updateInterval = 5f;
+    /// <summary>
+    /// Time since last update
+    /// </summary>
+    float elapsed = 0f;
+    
+    /// <summary>
+    /// Resolution of tile for the overlay
+    /// </summary>
+    public int xPixelsPerTile = 20;
+    public int yPixelsPerTile = 20;
+
+    /// <summary>
+    /// Internal storage of size of map
+    /// </summary>
+    public int xSize = 10;
+    public int ySize = 10;
+
+    /// <summary>
+    /// Script with user-defined lua valueAt functions
+    /// </summary>
+    Script script;
+
+    /// <summary>
+    /// 
+    /// </summary>
+    public string currentOverlay;
```

Comment:
If it's not a convention that the project uses (tell me if so), then no. By all means: properties are quite nice, but I wouldn't abuse them.